requirements/specifications/general/context-parameters.md

```diff
@@ -0,0 +1,303 @@
+**MUST** be before the value parameter, which itself **MUST** be last.
+
+#### 4.1.2. on<Property>Changed RPC generation
+
+When generating the on<Property>Changed notification for a property
+method, all the Context Parameters **MUST** be copied to the event
+parameters. The context parameters **MUST** be before the listen
+parameter, which itself **MUST** be last.
+
+#### 4.1.3. Temporal Set on<Item>Available / Unavailable RPC generation
+
+Generated Temporal Set events will treat the entire parameter list from
+the original temporal-set method as Context Parameters and copy them to
+the event parameters. The context parameters **MUST** be before the
+listen parameter, which itself **MUST** be last.
+
+#### 4.1.4. Event RPC Decoration
+
+All RPC methods tagged as event **MUST** have the listen parameter, of
+type boolean, automatically added when generating the final RPC. This
+parameter will be added at the end of the parameters list.
+
+All RPC methods tagged as event that have context parameters **MUST**
+have the result schema wrapped in an object. The original result schema
+**MUST** be moved to a property called data. Each of the context
+parameters **MUST** be copied to a property called context.
+
+So the following RPC event:
+
+```json
+{
+    "name": "onContextualEvent",
+    "tags": [
+        {
+            "name": "event"
+        }
+    ],
+    "params": [
+        {
+            "name": "a",
+            "required": true,
+            "schema": {
+                "type": "boolean"
+            }
+        },
+        {
+            "name": "b",
+            "required": false,
+            "schema": {
+                "type": "string"
+            }
+        },
+        {
+            "name": "c",
+            "required": false,
+            "schema": {
+                "type": "number"
+            }
+        }
+    ],
+    "result": {
+        "name": "result",
+        "schema": {
+            "type": "object",
+            "properties": {
+                "foo": {
+                    "type": "boolean"
+                }
+            }
+        }
+    }
+}
+```
+
+Would have its result transformed to:
+
+```json
+{
+    "name": "result",
+    "schema": {
+        "type": "object",
+        "properties": {
+            "data": {
+                "type": "object",
+                "properties": {
+                    "foo": {
+                        "type": "boolean"
+                    }
+                }
+            },
+            "context": {
+                "type": "object",
+                "properties": {
+                    "a": {
+                        "type": "boolean"
+                    },
+                    "b": {
+                        "type": "string"
+                    },
+                    "c": {
+                        "type": "number"
+                    }
+                },
+                "required": [
+                    "a"
```

Comment:
I thought every context parameter is required

Reply:
good catch, will fix.